relay-server/src/actors/processor.rs

```diff
@@ -1505,6 +1511,15 @@ impl EnvelopeProcessorService {
         }
     }
 
+    fn handle_encode_metrics(&self, message: EncodeMetrics) {
```

Comment:
I think some docs or a short comment would be nice here, why we can split logic in two different methods.

E.g. the processing one:
1. no partitioning required
2. we only have to do rate limits here
3. we only have to do cardinality limiting here

The PoP/non-processing one:
1. no rate limits
2. no cardinality limits
3. needs to partition